lib/connection/connection.js

```diff
@@ -38,175 +35,252 @@ var debugFields = [
+   * @param {number} [options.family=null] IP version for DNS lookup, passed down to Node's [`dns.lookup()` function](https://nodejs.org/api/dns.html#dns_dns_lookup_hostname_options_callback). If set to `6`, will only look for ipv6 addresses.
+   * @param {boolean} [options.keepAlive=true] TCP Connection keep alive enabled
+   * @param {number} [options.keepAliveInitialDelay=300000] Initial delay before TCP keep alive enabled
+   * @param {boolean} [options.noDelay=true] TCP Connection no delay
+   * @param {number} [options.connectionTimeout=30000] TCP Connection timeout setting
+   * @param {number} [options.socketTimeout=360000] TCP Socket timeout setting
+   * @param {boolean} [options.singleBufferSerializtion=true] Serialize into single buffer, trade of peak memory for serialization speed
+   * @param {boolean} [options.ssl=false] Use SSL for connection
+   * @param {boolean|function} [options.checkServerIdentity=true] Ensure we check server identify during SSL, set to false to disable checking. Only works for Node 0.12.x or higher. You can pass in a boolean or your own checkServerIdentity override function.
+   * @param {Buffer} [options.ca] SSL Certificate store binary buffer
+   * @param {Buffer} [options.crl] SSL Certificate revocation store binary buffer
+   * @param {Buffer} [options.cert] SSL Certificate binary buffer
+   * @param {Buffer} [options.key] SSL Key file binary buffer
+   * @param {string} [options.passphrase] SSL Certificate pass phrase
+   * @param {boolean} [options.rejectUnauthorized=true] Reject unauthorized server certificates
+   * @param {boolean} [options.promoteLongs=true] Convert Long values from the db into Numbers if they fit into 53 bits
+   * @param {boolean} [options.promoteValues=true] Promotes BSON values to native types where possible, set to false to only receive wrapper types.
+   * @param {boolean} [options.promoteBuffers=false] Promotes Binary BSON values to native Node Buffers.
+   */
+  constructor(socket, options) {
+    super();
+
+    options = options || {};
+    if (!options.bson) {
+      throw new TypeError('must pass in valid bson parser');
+    }
+
+    this.id = _id++;
+    this.options = options;
+    this.logger = Logger('Connection', options);
+    this.bson = options.bson;
+    this.tag = options.tag;
+    this.maxBsonMessageSize = options.maxBsonMessageSize || 1024 * 1024 * 16 * 4;
+
+    this.port = options.port || 27017;
+    this.host = options.host || 'localhost';
+    this.socketTimeout = typeof options.socketTimeout === 'number' ? options.socketTimeout : 360000;
+
+    // These values are inspected directly in tests, but maybe not necessary to keep around
+    this.keepAlive = typeof options.keepAlive === 'boolean' ? options.keepAlive : true;
+    this.keepAliveInitialDelay =
+      typeof options.keepAliveInitialDelay === 'number' ? options.keepAliveInitialDelay : 300000;
+    this.socketTimeout = typeof options.socketTimeout === 'number' ? options.socketTimeout : 360000;
```

Comment:
`this.socketTimeout` is also set in https://github.com/mongodb-js/mongodb-core/pull/396/files#diff-b472538a81cdbf2a40c0d2ccd5a7521dR94